Some rovinski cleanup rebased

oharboe opened the pull request:
Some commits that rebased cleanly from #1533

maliberty commented on src/drt/src/dr/FlexDR.cpp:
```diff
@@ -322,18 +322,8 @@ void FlexDR::initGCell2BoundaryPin()
-              } else {
-                hasRightBound = false;
-              }
+              bool hasLeftBound = bp.x() < leftBound;
+              bool hasRightBound = ep.x() >= rightBound;
```
these can be const now

maliberty commented on src/drt/src/dr/FlexDR.cpp:
```diff
@@ -354,18 +344,8 @@ void FlexDR::initGCell2BoundaryPin()
-              } else {
-                hasTopBound = false;
-              }
+              bool hasBottomBound = bp.y() < bottomBound;
+              bool hasTopBound = ep.y() >= topBound;
```
const

maliberty commented on src/drt/src/dr/FlexDR_init.cpp:
```diff
@@ -2542,6 +2535,7 @@ void FlexDRWorker::route_queue_init_queue(queue<RouteQueueEntry>& rerouteQueue)
     // ripup all nets and clear objs here
     // nets are ripped up during initNets()
     vector<drNet*> ripupNets;
+    ripupNets.reserve(nets_.size());
     for (auto& net : nets_) {
       ripupNets.push_back(net.get());
     }
```
This can just be

vector<drNet*> ripupNets(nets_);

maliberty commented on src/drt/src/dr/FlexGridGraph.cpp:
```diff
@@ -66,20 +69,23 @@ void FlexGridGraph::initGrids(
   // initialize all grids
   frMIdx xDim, yDim, zDim;
   getDim(xDim, yDim, zDim);
+  int capacity = xDim * yDim * zDim;
```
const

maliberty reviewed (approved):
After a review it is helpful if you don't force push, that way I can use the 'changes since last review' feature.

oharboe commented:
> After a review it is helpful if you don't force push, that way I can use the 'changes since last review' feature.

Ah. I'll leave the ```git commit --fixup```s in place next time.

maliberty commented:
It looks like these are subtly different in a way I didn't notice:
```
/OpenROAD/src/drt/src/dr/FlexDR_init.cpp:2537:35: error: no matching function for call to 'std::vector<fr::drNet*>::vector(std::vector<std::unique_ptr<fr::drNet> >&)'
```

I guess we'll have to go back to the prior version.  Sorry for the hassle.
